tests/unit/authenticationKey.test.ts

```diff
@@ -91,4 +91,13 @@ describe("AuthenticationKey", () => {
       "No supported authentication scheme for public key",
     );
   });
+
+  it("should serialize and deserialize correctly", () => {
+    const authKey = new AuthenticationKey({ data: ed25519.authKey });
+    const serializer = new Serializer();
+    authKey.serialize(serializer);
+    const deserializer = new Deserializer(serializer.toUint8Array());
+    const authKeyDeserialized = AuthenticationKey.deserialize(deserializer);
+    expect(authKeyDeserialized.data.toString()).toEqual(ed25519.authKey);
+  });
```

Comment:
FYI for a test like this, you should ensure that the data matches what you expect in bytes if it's supposed to be consumed by anything else.

You can serialize and deserialize something wrong, as long as it's symmetric :)

Reply:
Ah I kept forgetting about this. Will update soon